packages/ui/src/theme/__tests__/createTheme.test.ts

```diff
@@ -31,6 +31,180 @@ describe('@aws-amplify/ui', () => {
           'var(--amplify-colors-background-primary)'
         );
       });
+
+      it('should filter out deprecated pagination design tokens', () => {
+        const theme = {
+          name: 'test',
+          tokens: {
+            components: {
+              pagination: {
+                button: {
+                  _hover: {
+                    backgroundColor: {
+                      value: 'red',
+                    },
+                    color: {
+                      value: 'blue',
+                    },
+                  },
+                  _disabled: {
+                    color: {
+                      value: 'grey',
+                    },
+                  },
+                },
+              },
+            },
+          },
+        };
+        const { cssText, tokens } = createTheme(theme);
+        expect(cssText).toContain(
+          `--amplify-components-pagination-button-hover-background-color: ${tokens.components.pagination.button._hover?.backgroundColor.value}`
```

Comment:
What do you think of using `const hoverStateTokens = tokens.components.pagination.button._hover`? That will make the entire test more concise and crisp